Content.Server/Nyanotrasen/Shipyard/Systems/ShipyardSystem.Consoles.cs

```diff
@@ -0,0 +1,240 @@
+            {
+                ConsolePopup(args.Session, Loc.GetString("shipyard-console-invalid-vessel", ("vessel", args.Vessel)));
+                PlayDenySound(uid, component);
+                return;
+            }
+
+            if (!TryPurchaseVessel(bank, vessel, out var shuttle) || shuttle == null)
+            {
+                PlayDenySound(uid, component);
+                return;
+            }
+
+            _cargo.DeductFunds(bank, vessel.Price);
+            PlayConfirmSound(uid, component);
+            var newDeed = EnsureComp<ShuttleDeedComponent>(bank.Owner);
+
+            var newState = new ShipyardConsoleInterfaceState(
+                bank.Balance,
+                true);
+
+            _uiSystem.TrySetUiState(component.Owner, ShipyardConsoleUiKey.Shipyard, newState);
+            RegisterDeed(newDeed, shuttle);           
+        }
+
+        public void OnSellMessage(EntityUid uid, SharedShipyardConsoleComponent component, ShipyardConsoleSellMessage args)
+        {
+            
+            if (args.Session.AttachedEntity is not { Valid: true } player)
+            {
+                return;
+            }
+
+            var station = _station.GetOwningStation(uid);
+            var bank = GetBankAccount(station);
+
+            if (bank == null)
+                return;
+
+            if (!TryComp<ShuttleDeedComponent>(bank.Owner, out var deed) || deed == null || deed.ShuttleUid == null)
+            {
+                ConsolePopup(args.Session, Loc.GetString("shipyard-console-invalid-vessel"));
+                PlayDenySound(uid, component);
+                return;
+            }
+
+            if (!TrySellVessel(bank, deed.ShuttleUid, out var bill))
+            {
+                PlayDenySound(uid, component);
+                return;
+            };
+
+            _cargo.DeductFunds(bank, -bill);
+            PlayConfirmSound(uid, component);
+
+            ShipyardConsoleInterfaceState newState = new ShipyardConsoleInterfaceState(
+                    bank.Balance,
+                    true);
+
+            _uiSystem.TrySetUiState(component.Owner, ShipyardConsoleUiKey.Shipyard, newState);
+        }
+        private void OnConsoleUIOpened(EntityUid uid, SharedShipyardConsoleComponent component, BoundUIOpenedEvent args)
+        {
+            if (args.Session.AttachedEntity is not { Valid: true } player)
+                return;
+            var station = _station.GetOwningStation(component.Owner);
+            var bank = GetBankAccount(station);
+
+            if (bank == null)
+                return;
+
+            var newState = new ShipyardConsoleInterfaceState(
+                bank.Balance,
+                true);
+
+            _uiSystem.TrySetUiState(component.Owner, ShipyardConsoleUiKey.Shipyard, newState);
+        }
+
+        private void ConsolePopup(ICommonSession session, string text)
+        {
+            _popup.PopupCursor(text, session);
+        }
```

Comment:
Does it feel more intuitive to have the text popup at the cursor or at the entity? Usually these sort of messages popup on the console entity and not at the cursor.

Reply:
Yes this is holdover from ordering console, easy change